docs/docs/building-modules/09-module-interfaces-continued.md

```diff
@@ -0,0 +1,158 @@
+
+For example, suppose you have a service with two methods: `GetTest` and `GetTest2`. You can group these methods together under a `get` subcommand using the following code
+
+```
+autoCliOpts := &autocliv1.ModuleOptions{
+    Tx: &autocliv1.ServiceCommandDescriptor{
+        Service: myServiceDesc.ServiceName,
+        SubCommands: map[string]*autocliv1.ServiceCommandDescriptor{
+            "get": {
+                Service: myServiceDesc.ServiceName,
+                SubCommands: map[string]*autocliv1.ServiceCommandDescriptor{
+                    "test": {Service: myServiceDesc.ServiceName, Method: "GetTest"},
+                    "test2": {Service: myServiceDesc.ServiceName, Method: "GetTest2"},
+                },
+            },
+        },
+    },
+}
+
+
+```
+
+With this configuration, you can invoke the `GetTest` method by running `./app tx myservice get test`, and you can invoke the `GetTest2` method by running `./app tx myservice get test2`.
+
+#### Adding Custom Commands
+
+Although `autocli` can automatically generate CLI commands and flags based on your Protocol Buffer messages, you may want to add custom commands to your CLI application. To add a custom command, you can use the `AddCommand()` method provided by `cobra`. For instance, to add a custom `test` command to your CLI application, you can use the following code:
+
+```
+rootCmd := autocli.RootCmd(appOptions)
+rootCmd.AddCommand(&cobra.Command{
+    Use:   "test",
+    Short: "custom test command",
+    RunE: func(cmd *cobra.Command, args []string) error {
+        // custom command logic here
+        return nil
+    },
+})
+
+```
+
+This will add a new `test` command to your CLI application that can be invoked by running `./app test`.
+
+### Customising Flag Names
+
+By default, `autocli` generates flag names based on the names of the fields in your Protocol Buffer message. However, you can customise the flag names by providing a `namingOptions` parameter to the `Builder.AddMessageFlags()` method. This parameter allows you to specify custom names for flags based on the names of the message fields. For example, if you have a message with the fields `test` and `test1`, you can use the following naming options to customise the flags
+```
+options := autocliv1.RpcCommandOptions{ 
+	FlagOptions: map[string]*autocliv1.FlagOptions{ 
+		"test": { Name: "custom_name", }, 
+		"test1": { Name: "other_name", }, 
+		}, 
+	}
+```
+
+With these naming options, the generated flags for the `test` and `test1` fields will have custom names `--custom-name` and `--other-name`, respectively.
+
+### Conclusion
+
+`autocli` is a powerful tool for adding CLI interfaces to your Cosmos SDK-based applications. It allows you to easily generate CLI commands and flags from your Protocol Buffer messages, and provides many options for customising the behavior of your CLI application. With `autocli`, you can quickly create powerful CLI interfaces for your applications with minimal effort.
```

Comment:
I would be nice to put a link to hubl somewhere here too, showcasing what the SDK is doing for all chains.